### PR TITLE
Introduce partial C11 support

### PR DESCRIPTION
**Pull request by vit9696**
Implemented _Noreturn, _Static_assert, _Thread_local. Also fixed
tests with preprocessor on macOS.

### CODE REVIEW

File: tests/test_c_generator.py [hunk 1]
```diff
@@ -246,6 +250,13 @@ def test_issue246(self):
             int array[3] = {[0] = 0, [1] = 1, [1+1] = 2};
             ''')
 
+    def test_noreturn(self):
+        self._assert_ctoc_correct(r'''
+            _Noreturn int x(void) {
```
Comment (eliben) [inline tests/test_c_generator.py:255]: I'd like to see a standalone test for `_Noreturn` in `test_c_parser`. Just a small code snippet, and a test that verifies that the proper specifier is added to the function declaration.

File: utils/fake_libc_include/_fake_defines.h [hunk 2]
```diff
@@ -227,3 +225,15 @@
 #define __m512d_u int
 #define __m512i   int
 #define __m512i_u int
+
+/* C11 stdnoreturn.h defines */
+#define __noreturn_is_defined 1
+#define noreturn _Noreturn
+
+/* C11 threads.h defines */
+#define thread_local _Thread_local
```
Comment (eliben) [inline utils/fake_libc_include/_fake_defines.h:234]: Please add a specific parser test for `_Thread_local` too. The C-to-C tests are insufficient for this - they won't catch errors in building the AST in the parser.

Comment (eliben) [reply utils/fake_libc_include/_fake_defines.h:234]: ... and same comment for `static_assert`. Please test it both inside and outside a function, in `test_c_parser`.

File: pycparser/c_lexer.py [hunk 3]
```diff
@@ -109,19 +109,23 @@ def _make_tok_location(self, token):
...
     for keyword in keywords:
-        if keyword == '_BOOL':
-            keyword_map['_Bool'] = keyword
-        elif keyword == '_COMPLEX':
-            keyword_map['_Complex'] = keyword
-        else:
             keyword_map[keyword.lower()] = keyword
 
+    for keyword in keywords_new:
+        keyword_map[''.join([keyword[:2].upper(), keyword[2:].lower()])] = keyword
```
Comment (eliben) [inline pycparser/c_lexer.py:123]: Do you really need the `join` here, or would a `+` do?

**Review by eliben — CHANGES_REQUESTED** [review]

**Comment by vit9696** [comment]
All done.

**Comment by vit9696** [comment]
For your interest, the tests fail due to some GitHub Actions downtime. They pass on macOS for me.

**Review by eliben — APPROVED** [review]
Thanks for your patience in pushing this through. LGTM.
I have some minor styling comments, but will apply them myself after merging this PR - shortly.

**Pull request merged**